controller/deployments_urlprovider.go

```diff
@@ -91,130 +28,31 @@ var _ kubernetes.BaseURLProvider = (*tenantURLProvider)(nil)
-
-	if t.ID == nil {
-		log.Error(nil, map[string]interface{}{}, "app.UserService is malformed: no ID field")
-		return nil, errs.New("app.UserService is malformed: no ID field")
-	}
-
-	if t.Attributes == nil {
-		log.Error(nil, map[string]interface{}{
-			"tenant": *t.ID,
-		}, "app.UserService is malformed: no Attribute field ID=%s", *t.ID)
-		return nil, errs.Errorf("app.UserService is malformed: no Attribute field (ID=%s)", *t.ID)
-	}
-
-	if len(t.Attributes.Namespaces) == 0 {
-		log.Error(nil, map[string]interface{}{
-			"tenant": *t.ID,
-		}, "this tenant has no namespaces: %s", *t.ID)
-		return nil, errs.Errorf("app.UserService is malformed: no Namespaces (ID=%s)", *t.ID)
-	}
-
-	defaultNamespace := t.Attributes.Namespaces[0]
-	namespaceMap := make(map[string]*app.NamespaceAttributes)
-	for i, namespace := range t.Attributes.Namespaces {
-		namespaceMap[*namespace.Name] = t.Attributes.Namespaces[i]
-		if namespace.Type != nil && *namespace.Type == "user" {
-			defaultNamespace = namespace
-		}
-	}
-
-	defaultClusterURL := *defaultNamespace.ClusterURL
-
-	if len(proxyURL) != 0 {
-		// all non-metric API calls go via the proxy
-		defaultClusterURL = proxyURL
-	}
-
+// NewProxyURLProvider create a provider from a UserService object (exposed for testing)
+func NewProxyURLProvider(token string, proxyURL string) (kubernetes.BaseURLProvider, error) {
 	provider := &tenantURLProvider{
-		apiURL:     defaultClusterURL,
-		apiToken:   token,
-		tenant:     t,
-		namespaces: namespaceMap,
+		apiURL:   proxyURL,
+		apiToken: token,
 	}
 	return provider, nil
 }
 
-// NewTenantURLProviderFromTenant create a provider from a UserService object (exposed for testing)
-func NewTenantURLProviderFromTenant(t *app.UserService, token string, proxyURL string) (kubernetes.BaseURLProvider, error) {
-	return newTenantURLProviderFromTenant(t, token, proxyURL)
-}
-
-// GetEnvironmentMapping returns a map whose keys are environment names, and values are the Kubernetes namespaces
-// that represent those environments
-func (up *tenantURLProvider) GetEnvironmentMapping() map[string]string {
-	result := make(map[string]string)
-	// Exclude internal namespaces where the user cannot deploy applications
-
-	// Deployments API will receive requests by environment name (e.g. "run", "stage").
-	// These names correspond to the "type" attribute in Namespaces.
-	for envNS, attr := range up.namespaces {
-		envName := attr.Type
-		if envName == nil || len(*envName) == 0 {
-			log.Error(nil, map[string]interface{}{
-				"namespace": envNS,
-			}, "namespace has no type")
-		} else {
-			result[*envName] = envNS
-		}
-	}
-	return result
-}
-
-// Types of namespaces where the user does not deploy applications
-var internalNamespaceTypes = map[string]struct{}{"user": {}, "che": {}, "jenkins": {}}
-
-// CanDeploy returns true if the environment type provided can be deployed to as part of a pipeline
-func (up *tenantURLProvider) CanDeploy(envType string) bool {
```

Comment:
We need this one too, to filter out environments we want to hide like che and jenkins.